src/main/java/org/openstreetmap/atlas/checks/validation/RoadNameGapCheck.java

```diff
@@ -0,0 +1,134 @@
+package org.openstreetmap.atlas.checks.validation;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.atlas.predicates.TagPredicates;
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * This check identifies Edges with no name Tag that are between two other Edges with the same name
+ * Tag, OR the Edge has a name Tag but does not equal the name Tag of the Edges that it is between.
+ *
+ * @author sugandhimaheshwaram
+ */
+public class RoadNameGapCheck extends BaseCheck
+{
+
+    // You can use serialver to regenerate the serial UID.
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public RoadNameGapCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge && Edge.isMasterEdgeIdentifier(object.getIdentifier())
+                && TagPredicates.IS_HIGHWAY_NOT_LINK_TYPE.test(object)
+                && TagPredicates.VALID_HIGHWAY_TAG.test(object)
+                && HighwayTag.isCarNavigableHighway(object) && !JunctionTag.isRoundabout(object);
```

Comment:
Since all tags checked in ```VALID_HIGHWAY_TAG``` are contained in the set of car navigable highways, we could just call ```TagPredicates.VALID_HIGHWAY_TAG.test(object)``` without ```HighwayTag.isCarNavigableHighway(object)```

Reply:
Also, would we want to check for the missing name tag here? For ```object```

Reply:
The requirement mentioned about all the valid tags along with car navigable highway so I was using the default method. Will take it out. Thanks.